src/main/java/org/openrewrite/maven/AbstractRewriteMojo.java

```diff
@@ -262,15 +267,22 @@ protected LargeSourceSet loadSourceSet(Path repositoryRoot, Environment env, Exe
 
     protected List<Result> runRecipe(Recipe recipe, LargeSourceSet sourceSet, ExecutionContext ctx) {
         getLog().info("Running recipe(s)...");
-        return recipe.run(sourceSet, ctx).getChangeset().getAllResults().stream()
-                .filter(source -> {
-                    // Remove ASTs originating from generated files
-                    if (source.getBefore() != null) {
-                        return !source.getBefore().getMarkers().findFirst(Generated.class).isPresent();
-                    }
-                    return true;
-                })
-                .collect(toList());
+        RecipeRun recipeRun = recipe.run(sourceSet, ctx);
+
+        if (exportDatatables) {
+            String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd_HH-mm-ss-SSS"));
+            Path datatableDirectoryPath = Paths.get("target", "rewrite", "datatables", timestamp);
```

Comment:
I wonder if we should include the timestamp here, or if we should just use stable predictable paths that get overwritten. Any thoughts on that yourself?

Reply:
I went back and forth on this as well. I think keeping timestamps is valuable, it provides a bit more flexibility and prevents accidental overwriting in cases where someone runs two different recipes that happen to produce the same datatable (i.e. the RecipeRunStats datatable for example). It does add a bit more bulk, but it is also in the target directory which is pretty ephemeral.

Reply:
then indeed let's keep the timestamps. I feel like either option would get criticized anyway, no matter which we choose. ;)